tabrepo/repository/evaluation_repository_collection.py

```diff
@@ -0,0 +1,258 @@
+        super().__init__(zeroshot_context=zeroshot_context, config_fallback=config_fallback)
+
+    def _update_mapping(self):
+        combinations = self._generate_dataset_fold_config_combinations(repos=self.repos)
+        return self._combination_mapping_to_repo_index(combinations=combinations, overlap=self.overlap)
+
+    def goes_where(self, dataset: str, fold: int, config: str) -> int | None:
+        """
+        Returns the repo idx containing the specified config in a given dataset fold. Returns None if no such repo exists.
+        """
+        return self._mapping.get((dataset, fold, config), None)
+
+    def predict_test_multi(self, dataset: str, fold: int, configs: List[str] = None, binary_as_multiclass: bool = False) -> np.ndarray:
+        return self._predict_multi(predict_func="predict_test_multi", dataset=dataset, fold=fold, configs=configs, binary_as_multiclass=binary_as_multiclass)
+
+    def predict_val_multi(self, dataset: str, fold: int, configs: List[str] = None, binary_as_multiclass: bool = False) -> np.ndarray:
+        return self._predict_multi(predict_func="predict_val_multi", dataset=dataset, fold=fold, configs=configs, binary_as_multiclass=binary_as_multiclass)
+
+    def _predict_multi(self, predict_func: str, dataset: str, fold: int, configs: List[str] = None, binary_as_multiclass: bool = False) -> np.ndarray:
+        if configs is None:
+            configs = self.configs()
+
+        repo_map = defaultdict(list)
+        repo_idx_map = defaultdict(list)
+        predict_map = dict()
+        config_idx_lst = []
+        for i, config in enumerate(configs):
+            repo_idx = self.goes_where(dataset=dataset, fold=fold, config=config)
+            if repo_idx is None:
+                if self._config_fallback is None:
+                    raise ValueError(f"The following combination is not present in this repository: (dataset='{dataset}', fold={fold}, config='{config}')")
+                # get fallback
+                repo_idx = self.goes_where(dataset=dataset, fold=fold, config=self._config_fallback)
+                if repo_idx is None:
+                    raise ValueError(
+                        f"The following combination is not present in this repository: (dataset='{dataset}', fold={fold}, config='{config}')"
+                        f"\nAdditionally, the fallback config is not present in (dataset='{dataset}', fold={fold}, config='{self._config_fallback}')"
+                    )
+                repo_map[repo_idx].append(self._config_fallback)
+            else:
+                repo_map[repo_idx].append(config)
+            repo_idx_map[repo_idx].append(i)
+            config_idx_lst.append(repo_idx)
+
+        for repo_idx, config_lst in repo_map.items():
+            f = getattr(self.repos[repo_idx], predict_func)
+            predict_map[repo_idx] = f(dataset=dataset, fold=fold, configs=config_lst, binary_as_multiclass=binary_as_multiclass)
+
+        shape = predict_map[repo_idx].shape[1:]
```

Comment:
Added the following inline comment:

```
        # predict_map[repo_idx] has shape
        #  (n_configs, n_rows, n_classes) if multiclass classification
        #  (n_configs, n_rows, n_classes) if binary classification and binary_as_multiclass=True
        #  (n_configs, n_rows) otherwise
        # We ignore the first dimension because we need all configs in the final result,
        # and each repo could only have a subset of the configs.
```